### PR TITLE
Print raw error msg if requested + Print session filename

### PR DESCRIPTION
**Pull request by dluc**
- Fix: When using '--raw' then print the raw error message (using Util to handle circular references in the error object)
- Improve: Print name of the file where the auth session is stored, so that a user is aware of credentials stored locally (e.g. to remove them if required)

### CODE REVIEW

**Comment by azurecla** [comment]
Hi __@dluc__, I'm your friendly neighborhood Azure Pull Request Bot (You can call me AZPRBOT). Thanks for your contribution!
    <span>
        In order for us to evaluate and accept your PR, we ask that you sign a contribution license agreement. It's all electronic and will take just minutes. I promise there's no faxing. https://cla.azure.com.
    </span>

TTYL, AZPRBOT;

File: tools/iothub-explorer/package.json [hunk 1]
```diff
@@ -15,7 +15,8 @@
     "colors-tmpl": "^1.0.0",
     "nopt": "^3.0.4",
     "prettyjson": "^1.1.3",
-    "uuid": "^2.0.1"
+    "uuid": "^2.0.1",
+    "util": "0.10.3"
```
Comment (pierreca) [inline tools/iothub-explorer/package.json]: not too sure if this needs to be added here, I thought util was a native node module.

File: tools/iothub-explorer/iothub-explorer.js [hunk 2]
```diff
@@ -95,6 +96,7 @@ if (!connString) {
   var loc = configLoc();
   try {
     sas = fs.readFileSync(loc.dir + '/' + loc.file, 'utf8');
+    console.log(colorsTmpl('\n{green}Using session file: '+loc.dir + '/' + loc.file + '{/green}'));
```
Comment (pierreca) [inline tools/iothub-explorer/iothub-explorer.js]: spacing between + and loc.dir is off

File: tools/iothub-explorer/iothub-explorer.js [hunk 3]
```diff
@@ -20,6 +20,7 @@ var ConnectionString = require('azure-iothub').ConnectionString;
 var SharedAccessSignature = require('azure-iothub').SharedAccessSignature;
 var anHourFromNow = require('azure-iot-common').anHourFromNow;
 var DeviceSAS = require('azure-iot-device').SharedAccessSignature;
+var Util = require('util');
```
Comment (pierreca) [inline tools/iothub-explorer/iothub-explorer.js]: I may be mistaking this for something else, but if this is the native util package, our convention is to use lowercase for the variable name

File: tools/iothub-explorer/iothub-explorer.js [hunk 4]
```diff
@@ -321,6 +323,9 @@ function serviceError(err) {
     }
     console.log(colorsTmpl('\n{bold}{red}Error{/red}{/bold}' + message));
   }
+  else {
```
Comment (pierreca) [inline tools/iothub-explorer/iothub-explorer.js]: the else should be on the same line as the closing brace

File: tools/iothub-explorer/iothub-explorer.js [hunk 5]
```diff
@@ -321,6 +323,9 @@ function serviceError(err) {
...
+    console.log(colorsTmpl('\n{bold}{red}Raw error message: ' + Util.inspect(err) + '{/red}{/bold}'));
```
Comment (pierreca) [inline tools/iothub-explorer/iothub-explorer.js]: To be consistent with the non-raw error message, I would only show "Raw error message" in red, and maybe the rest using the regular console font.

**Comment by pierreca** [comment]
@dluc Thanks a lot for the PR, I'm all for it. I found a couple points to nitpick on regarding code formatting but other than that, LGTM.
Per our contribution guidelines though, you should resubmit this PR against our develop branch so that we can run it through our gated builds. we never check code directly into the master branch, which is used only to show the current state of the released code. (the develop branch is where the work happen and is merged to master on a regular basis as we released new packages).

Thanks!

**Comment by tameraw** [comment]
@dluc - If you're still interested in this PR, can you please address the comments, sign the CLA and change the base to develop.

**Comment by tameraw** [comment]
@dluc 
Please address comments and resubmit against the iothub-explorer repo.
https://github.com/azure/iothub-explorer
Thanks.

**Pull request closed without merging**